app/operations/memberships/create.rb

```diff
@@ -0,0 +1,26 @@
+module Memberships
+  class Create < Operation
+    string :join_token, default: nil
+
+    hash :links do
+      integer :user
+      integer :user_group
+    end
+
+    def execute
+      raise Unauthenticated unless api_user.logged_in?
+      raise Unauthorized unless user_group.verify_join_token(join_token)
+      raise Unauthorized unless user.id == api_user.id
+
+      Membership.create! user: api_user.user, user_group: user_group, state: :active
+    end
+
+    def user
+      User.find(links["user"])
```

Comment:
These will raise if not found. Just checking we don't want to catch this and propagate up to the api controller.